matter/src/mdns.rs

```diff
@@ -72,31 +73,62 @@ impl Mdns {
     /// Set mDNS service specific values
     /// Values like vid, pid, discriminator etc
     // TODO: More things like device-type etc can be added here
-    pub fn set_values(&self, vid: u16, pid: u16, discriminator: u16) {
+    pub fn set_values(&self, vid: u16, pid: u16, discriminator: u16, device_name: &str) {
         let mut inner = self.inner.lock().unwrap();
         inner.vid = vid;
         inner.pid = pid;
         inner.discriminator = discriminator;
+        inner.device_name = device_name.chars().take(32).collect();
     }
 
     /// Publish a mDNS service
     /// name - is the service name (comma separated subtypes may follow)
     /// mode - the current service mode
+    #[allow(clippy::needless_pass_by_value)]
     pub fn publish_service(&self, name: &str, mode: ServiceMode) -> Result<SysMdnsService, Error> {
         match mode {
             ServiceMode::Commissioned => {
                 sys_publish_service(name, "_matter._tcp", MATTER_PORT, &[])
             }
             ServiceMode::Commissionable => {
                 let inner = self.inner.lock().unwrap();
-                let short =
-                    (inner.discriminator & SHORT_DISCRIMINATOR_MASK) >> SHORT_DISCRIMINATOR_SHIFT;
+                let short = compute_short_discriminator(inner.discriminator);
                 let serv_type = format!("_matterc._udp,_S{},_L{}", short, inner.discriminator);
 
                 let str_discriminator = format!("{}", inner.discriminator);
-                let txt_kvs = [["D", &str_discriminator], ["CM", "1"]];
+                let txt_kvs = [
+                    ["D", &str_discriminator],
+                    ["CM", "1"],
+                    ["DN", &inner.device_name],
+                    ["VP", &format!("{}+{}", inner.vid, inner.pid)],
+                    ["SII", "5000"], /* Sleepy Idle Interval */
+                    ["SAI", "300"],  /* Sleepy Active Interval */
+                    ["T", "1"],      /* TCP supported */
```

Comment:
We don't support TCP, so we should not have this?

Reply:
Removed the "T" tag.